bigtable/google/cloud/bigtable/table.py

```diff
@@ -184,15 +185,21 @@ def create(self):
                                                 table_id=self.table_id,
                                                 table={})
 
-    def exist(self):
-        """Checks whether a table with given table ID exist
-        or not for given Bigtable instance.
+    def exists(self):
+        """Check whether the table exists.
+
+        :rtype: bool
+        :returns: True if the table exists, else False.
         """
-        instance = self._instance
-        for table in instance.list_tables():
-            if self.table_id == table.table_id:
-                return True
-        return False
+        table_client = self._instance._client.table_admin_client
+        try:
+            table_client.get_table(name=self.name)
+        except NotFound:
+            return False
+        except StopIteration:
+            return False
```

Comment:
I don't see how `google.cloud.bigtable_admin_v2.gapic.bigtable_table_admin_client.BigtableTableAdminClient.get_table` would ever raise a `StopIteration`:  it is defined to return a single `google.cloud.bigtable_admin_v2.types.Table` message.

Reply:
Agree with you. I also worried about that. But get_table gives StopIteration. Can you check my unittest test_exists() populating mock environment correctly.